Bug: Molecule and Molecule Plugins packages missing from Tox Integration Environment install_deps

## Summary

When using `tox-ansible` version `26.3.0` for integration tests on a collection which follows the [User Guide - Testing Molecule Scenarios](https://docs.ansible.com/projects/tox-ansible/user_guide/#testing-molecule-scenarios) method. 

In the `tox-ansible` `integration-*` environments `install_deps` phase the `molecule` and `molecule-plugins` packages are not installed causing `pytest-ansible` to fail:

```
uv run tox --ansible --conf tox-ansible.ini -e integration-py3.14-2.20
integration-py3.14-2.20: install_deps> .venv/bin/uv pip install 'ansible-compat>=25.11.0' 'ansible-dev-environment>=26.2.0' 'pytest-ansible>=v4.1.1' 'pytest-xdist>=3.4.0' 'pytest>=7.4.3'
integration-py3.14-2.20: commands[0]> python3 -m pytest --ansible-unit-inject-only ./tests/integration
=================================================================================================================================== test session starts ===================================================================================================================================
platform linux -- Python 3.14.3, pytest-9.0.3, pluggy-1.6.0
cachedir: .tox/integration-py3.14-2.20/.pytest_cache
ansible: 2.20.4
configfile: pyproject.toml
plugins: ansible-26.4.0, xdist-3.8.0, anyio-4.13.0
collected 0 items / 1 error                                                                                                                                                                                                                                                               

========================================================================================================================================= ERRORS ==========================================================================================================================================
_________________________________________________________________________________________________________________ ERROR collecting tests/integration/test_integration.py __________________________________________________________________________________________________________________
.tox/integration-py3.14-2.20/lib64/python3.14/site-packages/pluggy/_hooks.py:512: in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.tox/integration-py3.14-2.20/lib64/python3.14/site-packages/pluggy/_manager.py:120: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.tox/integration-py3.14-2.20/lib64/python3.14/site-packages/_pytest/python.py:249: in pytest_pycollect_makeitem
    return list(collector._genfunctions(name, obj))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.tox/integration-py3.14-2.20/lib64/python3.14/site-packages/_pytest/python.py:475: in _genfunctions
    self.ihook.pytest_generate_tests.call_extra(methods, dict(metafunc=metafunc))
.tox/integration-py3.14-2.20/lib64/python3.14/site-packages/pluggy/_hooks.py:573: in call_extra
    return self._hookexec(self.name, hookimpls, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.tox/integration-py3.14-2.20/lib64/python3.14/site-packages/pluggy/_manager.py:120: in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.tox/integration-py3.14-2.20/lib64/python3.14/site-packages/pytest_ansible/plugin.py:407: in pytest_generate_tests
    pytest.exit("molecule not installed or found.")
E   _pytest.outcomes.Exit: molecule not installed or found.
================================================================================================================================= short test summary info =================================================================================================================================
ERROR tests/integration/test_integration.py - _pytest.outcomes.Exit: molecule not installed or found.
```

Current workaround:

```ini
[testenv]
commands_pre =
    integration: uv pip install 'molecule' 'molecule-plugins[docker]'
```

`molecule-plugins` being installed I guess is up for discussion, but IMHO at a minimum if we want to pattern to work we need `molecule` installed.

--- a/src/tox_ansible/plugin.py
+++ b/src/tox_ansible/plugin.py
@@ -675,6 +675,8 @@ def conf_deps(env_conf: EnvConfigSet, test_type: str) -> str:  # noqa: ARG001
         deps.append("ansible-dev-environment>=26.2.0")
         if test_type in ("integration", "unit"):
             deps.extend(OUR_DEPS)
+            if test_type == "integration":
+                deps.append("molecule>=26.4.0")
             try:
                 with (cwd / "test-requirements.txt").open() as fileh:
                     deps.extend(fileh.read().splitlines())

--- a/src/tox_ansible/plugin.py
+++ b/src/tox_ansible/plugin.py
@@ -677,21 +677,12 @@ def conf_deps(env_conf: EnvConfigSet, test_type: str) -> str:  # noqa: ARG001
             deps.extend(OUR_DEPS)
             if test_type == "integration":
                 deps.append("molecule>=26.4.0")
-            try:
-                with (cwd / "test-requirements.txt").open() as fileh:
-                    deps.extend(fileh.read().splitlines())
-            except FileNotFoundError:
-                pass
-            try:
-                with (cwd / "requirements-test.txt").open() as fileh:
-                    deps.extend(fileh.read().splitlines())
-            except FileNotFoundError:
-                pass
-            try:
-                with (cwd / "requirements.txt").open() as fileh:
-                    deps.extend(fileh.read().splitlines())
-            except FileNotFoundError:
-                pass
+            for req_file in ("test-requirements.txt", "requirements-test.txt", "requirements.txt"):
+                try:
+                    with (cwd / req_file).open() as fileh:
+                        deps.extend(fileh.read().splitlines())
+                except FileNotFoundError:
+                    pass
 
     return "\n".join(deps)
 

--- a/src/tox_ansible/plugin.py
+++ b/src/tox_ansible/plugin.py
@@ -681,7 +681,7 @@ def conf_deps(env_conf: EnvConfigSet, test_type: str) -> str:  # noqa: ARG001
                 try:
                     with (cwd / req_file).open() as fileh:
                         deps.extend(fileh.read().splitlines())
-                except FileNotFoundError:
+                except FileNotFoundError:  # noqa: PERF203
                     pass
 
     return "\n".join(deps)

--- a/tests/unit/test_plugin.py
+++ b/tests/unit/test_plugin.py
@@ -512,6 +512,30 @@ def test_conf_deps(tmp_path: Path) -> None:
         assert "requirement" in result
         assert "requirement-test" in result
         assert "github.com/ansible/ansible" not in result
+        assert "molecule" not in result
+
+
+def test_conf_deps_integration(tmp_path: Path) -> None:
+    """Test the conf_deps function includes molecule for integration tests.
+
+    Args:
+        tmp_path: Pytest fixture.
+    """
+    ini_file = tmp_path / "tox.ini"
+    ini_file.touch()
+    source = discover_source(ini_file, None)
+
+    with working_directory(tmp_path):
+        conf = Config.make(
+            Parsed(work_dir=tmp_path, override=[], config_file=ini_file, root_dir=tmp_path),
+            pos_args=[],
+            source=source,
+            extra_envs=[],
+        ).get_env("integration-py3.14-2.19")
+
+        result = conf_deps(env_conf=conf, test_type="integration")
+        assert "ansible-dev-environment>=26.2.0" in result
+        assert "molecule>=26.4.0" in result
 
 
 def test_conf_deps_sanity(tmp_path: Path) -> None:

